src/logic/player.h

```diff
@@ -588,6 +590,7 @@ class Player {
 	bool is_attack_forbidden(PlayerNumber who) const;
 
 	const std::string pick_shipname();
+	const std::string pick_portname();
```

Comment:
I don't know where to put this, but I don't want to forget: So names are removed from candidate list on assignment, and not re-added when the name is changed.

I guess that's not only hard to track, but also makes sense when the player changes it, because we can assume they wouldn't have changed the name if they didn't mind having it.

But on loading, you first do an `init()` for the portdock, which picks a name, which removes that name from the list, then you assign the loaded name. I think map objects are loaded after the player info, ie. after restoring the list of remaining names, so an unused random name will be removed from the list.

Reply:
init is not called during loading so this is not a problem. I added an assert that the name is empty before loading.

Reply:
Sorry, my mistake. It's `loader->init()`.